tests/publiccloud/check_registercloudguest.pm

```diff
@@ -39,80 +39,75 @@ sub run {
         # all BYOS related checks. So we just regestering system and going further
         registercloudguest($instance);
     } elsif (is_byos()) {
-        if ($instance->run_ssh_command(cmd => 'sudo zypper lr', proceed_on_failure => 1) !~ /No repositories defined/gm) {
+        if ($instance->ssh_script_output(cmd => 'sudo zypper lr') !~ /No repositories defined/gm) {
             die 'The BYOS instance should be unregistered and report "Warning: No repositories defined.".';
         }
 
-        if ($instance->run_ssh_command(cmd => 'sudo systemctl is-enabled guestregister.service', proceed_on_failure => 1) !~ /disabled/) {
-            die('guestregister.service is not disabled');
-        }
+        $instance->ssh_assert_script_run(cmd => '! sudo systemctl is-enabled guestregister.service', fail_message => 'guestregister.service is not disabled');
```

Comment:
again `systemctl is-enabled guestregister.service` may return non-zero exit code not only because service is disabled . systemctl crash will also be treated by this version of the test as positive

Reply:
> If it is or if the state cannot be determined (i.e. your other non-zero exit codes) then the test should fail

exactly ! totally agree we should fail and we won't  because "! sudo systemctl is-enabled guestregister.service" will give you 0 for ANY none-zero value due to "!"

Reply:
Sorry my bad. Yes Anton is right here. @pdostal we need to really check or for the `1` value of for the `disabled` output, otherwise any other issues are undetected. I would leave this as close as possible to the original state

```suggestion
        if ($instance->ssh_script_output(cmd => 'sudo systemctl is-enabled guestregister.service') !~ /disabled/) die('guestregister.service is not disabled');
```

And yes, in principle this should not make a difference, but since the topic was raised ...